### PR TITLE
Add special handling for Implements and Inherts lists

### LINKED ISSUE #5097: CA1001 Fixer generates incorrect code
image
As you can see.  When you run the fixer with a class that already implements an interface, it adds a 
```VB
, IDisposable
```
line to the code file.  The only problem is that the Implements statement does not allow multiple lines unless the "," is on the end of the line like this

```VB
 Implements IExporter,
            IDisposable
```

### PR DESCRIPTION
**Pull request by mattwar**
Fixes #5097 

Changes how vb syntax generator deals with the AddInterfaceType API, preserving the trivia around the inherits and implements clause. 

@dpoeschl @DustinCampbell @rchande @Pilchie @jaredpar please review

### CODE REVIEW

File: src/Workspaces/VisualBasicTest/CodeGeneration/SyntaxGeneratorTests.vb [hunk 1]
```diff
@@ -2786,6 +2786,77 @@ End Interface</x>.Value)
...
+        <WorkItem(5097, "https://github.com/dotnet/roslyn/issues/5097")>
+        Public Sub TestAddInterfaceWithEOLs()
+            Dim classC = SyntaxFactory.ParseCompilationUnit("
+Public Class C
+End Class").Members(0)
+
+            VerifySyntaxRaw(Of ClassBlockSyntax)(
+                _g.AddInterfaceType(classC, _g.IdentifierName("X")), "
+Public Class C
+ImplementsXEnd Class")
```
Comment (DustinCampbell) [inline src/Workspaces/VisualBasicTest/CodeGeneration/SyntaxGeneratorTests.vb]: This doesn't seem like a great result.

File: src/Workspaces/VisualBasicTest/CodeGeneration/SyntaxGeneratorTests.vb [hunk 2]
```diff
@@ -2786,6 +2786,77 @@ End Interface</x>.Value)
...
+
+            Dim interfaceI = SyntaxFactory.ParseCompilationUnit("
+Public Interface I
+End Interface").Members(0)
+
+            VerifySyntaxRaw(Of InterfaceBlockSyntax)(
+                _g.AddInterfaceType(interfaceI, _g.IdentifierName("X")), "
+Public Interface I
+InheritsXEnd Interface")
```
Comment (DustinCampbell) [inline src/Workspaces/VisualBasicTest/CodeGeneration/SyntaxGeneratorTests.vb]: This doesn't either.

File: src/Workspaces/VisualBasicTest/CodeGeneration/SyntaxGeneratorTests.vb [hunk 3]
```diff
@@ -2786,6 +2786,77 @@ End Interface</x>.Value)
...
+    Implements X
+    Implements Y
+End Class").Members(0)
+
+            VerifySyntaxRaw(Of ClassBlockSyntax)(
+                _g.AddInterfaceType(classCXY, _g.IdentifierName("Z")), "
+Public Class C
+    Implements X
+    Implements Y
+ImplementsZEnd Class")
```
Comment (DustinCampbell) [inline src/Workspaces/VisualBasicTest/CodeGeneration/SyntaxGeneratorTests.vb]: Shouldn't adding a new Implements statement include a statement terminator?

File: src/Workspaces/VisualBasicTest/CodeGeneration/SyntaxGeneratorTests.vb [hunk 4]
```diff
@@ -2786,6 +2786,77 @@ End Interface</x>.Value)
...
+    Inherits X
+    Inherits Y
+End Interface").Members(0)
+
+            VerifySyntaxRaw(Of InterfaceBlockSyntax)(
+                _g.AddInterfaceType(interfaceIXY, _g.IdentifierName("Z")), "
+Public Interface I
+    Inherits X
+    Inherits Y
+InheritsZEnd Interface")
```
Comment (DustinCampbell) [inline src/Workspaces/VisualBasicTest/CodeGeneration/SyntaxGeneratorTests.vb]: Ditto

File: src/Workspaces/VisualBasicTest/CodeGeneration/SyntaxGeneratorTests.vb (hunk 1 above)
Comment (mattwar) [inline src/Workspaces/VisualBasicTest/CodeGeneration/SyntaxGeneratorTests.vb]: These tests a asserting the preformatted state after the operation is applied.  None of the SyntaxGenerator API add/manipulate trivia. All scenarios using SyntaxGenerator eventually hit the formatter.

**Pull request merged**